### DISCUSSION TITLE
Is this project maintained

### DISCUSSION THREAD

**Discussion by eliassal** [security-code-scan/security-code-scan#255] [General]
Hi, can you please tell me if this project maintained? I cloned the repo vs2019 and opened it in VS 2022 but it is not compiling because of the netcoreapp 2.0, how this can be resolved?

**Comment by JarLob** [comment]
It compiles both on my machine and in [the cloud](https://github.com/security-code-scan/security-code-scan/runs/7242999832?check_suite_focus=true) (you may look into [the workflow ](https://github.com/security-code-scan/security-code-scan/blob/vs2019/.github/workflows/build.yml) to check what it is needed to build it). It targets .NET Standard 2.0 and .NET 4.8, so most probably your local installation lacks these components.

**Comment by eliassal** [comment]
Thanks, I reopened VS2022 and did Nuget rsore and was able to compile (I have already netapp2).
I ran my compiled security-scan.exe on 2 SLNs, one worked correctly, a 2nd raised the following

![SecScan](https://user-images.githubusercontent.com/8447864/183375722-7a69c02f-0144-45a9-9442-1b81cbf16c37.jpg)


which I am not sure is it related to the securit scan tool or the project itself

here is a snapshot for the sln that went through correctly
![SecScan_002](https://user-images.githubusercontent.com/8447864/183376075-5b74bd43-5513-48ce-894b-53a68e89fa74.jpg)


By the way, is it possible to run the code against *.csproj or vbproj file?if yes, how?  I use heavily VSC which does not create sln file.
Thanks in advance

**Comment by eliassal** [comment]
Also, if you allow, can you please point me to a document or a tutorial on how to use this scan tool in an Azure Build/Release Pipeline? Thanks again

**Comment by JarLob** [comment]
> Also, if you allow, can you please point me to a document or a tutorial on how to use this scan tool in an Azure Build/Release Pipeline? Thanks again

Did you check already the documentation from https://security-code-scan.github.io/? SCS doesn't have specific integration for Azure CI, but it is a command line tool which you may call in multiple ways.

**Comment by JarLob** [comment]
> By the way, is it possible to run the code against *.csproj or vbproj file?if yes, how? I use heavily VSC which does not create sln file. Thanks in advance

The standalone tool calls [OpenSolutionAsync](https://github.com/security-code-scan/security-code-scan/blob/08738df93268564caab45a10f8423726dcddb064/SecurityCodeScan.Tool/Program.cs#L296), but the [MSBuildWorkspace](https://github.com/dotnet/roslyn/blob/afef8332b40c289ff99140adff92aa2d07776053/src/Workspaces/Core/MSBuild/MSBuild/MSBuildWorkspace.cs#L222) has `OpenProjectAsync` too, so I guess it is possible to implement.
BTW, the standalone tool is not the only way to invoke SCS. It can be added to a specific project as a NuGet package.

**Comment by eliassal** [comment]
OK Thanks, the problem is that I have so many tasks and  I have no time to implement for the time being. Regards

**Reply by JarLob** [reply]
FYI, it was implemented in 5.6.4

**Reply by eliassal** [reply]
Jarlob, can you please tell me how? Thanks in advance

**Reply by JarLob** [reply]
Update to the latest version. Pass csproj instead of sln.

**Comment by eliassal** [comment]
So many thanks for your prompt hep and support. It works like a charm with csproj. Great work
Canyou please tell me what is the difference between this tool and SonarQube? Do they do the same thing, or in other words, does it satisfy the same objectif?

**Reply by JarLob** [reply]
SCS is similar to SonarQube, Acunetix, Coverity and others in a sense that it is [SAST](https://en.wikipedia.org/wiki/Static_application_security_testing) tool.

**Comment by eliassal** [comment]
Thanks got it.
